src/Symfony/Bundle/SecurityBundle/Resources/config/security.php

```diff
@@ -65,6 +66,12 @@
             ])
         ->alias(AuthorizationCheckerInterface::class, 'security.authorization_checker')
 
+        ->set('security.user_authorization_checker', UserAuthorizationChecker::class)
+            ->args([
+                service('security.access.decision_manager'),
+            ])
+        ->alias(UserAuthorizationChecker::class, 'security.user_authorization_checker')
```

Comment:
We need a functional test that makes sure that the service is wired properly. We should also have a test that makes sure, the new `UserAuthorizationChecker` checks the permissions of the given user instead of the authenticated user.

Reply:
Added a functional test for checking that the correct user is being checked against. I don't see a test for autowiring for the current `UserAuthorizationChecker`. Do you have an example of what you're wanting for that test?

Reply:
I didn't say "_auto_-wiring". 😉  Your test is fine. 👍🏻